netbout-web/src/main/java/com/netbout/dynamo/DyMessages.java

```diff
@@ -202,7 +203,11 @@ public long unread() throws IOException {
             )
             .where(DyFriends.HASH, Conditions.equalTo(this.bout))
             .where(DyFriends.RANGE, Conditions.equalTo(this.self))
-            .iterator().next();
+            .iterator();
+        if (!iterator.hasNext()) {
+            throw new Inbox.BoutNotFoundException(this.bout);
```

Comment:
@2686747 I think here `HttpException` (404) is fine and in line with what the architect said ( mentioned somewhere that this situation should work out like a missing bout ).
And yes with a 404 the test is easy imo, I think that's why I was such a pain about that from the get go, sorry ;)